src/patterns/ec2-app.test.ts

```diff
@@ -222,6 +222,43 @@ describe("the GuEC2App pattern", function () {
     ).toThrowError();
   });
 
+  it("can configure ASG scaling by stage if desired", function () {
+    const stack = simpleGuStackForTesting();
+    const app = "test-gu-ec2-app";
+    new GuEc2App(stack, {
+      applicationPort: GuApplicationPorts.Node,
+      access: { scope: AccessScope.PUBLIC },
+      app: app,
+      certificateProps: getCertificateProps(),
+      monitoringConfiguration: { noMonitoring: true },
+      userData: "",
+      scaling: {
+        CODE: { minimumInstances: 3 },
+        PROD: { minimumInstances: 5, maximumInstances: 12 },
+      },
+    });
+
+    expect(stack).toHaveResource("AWS::AutoScaling::AutoScalingGroup", {
```

Comment:
Personally I wouldn't bother with this assertion here, as it feels like a responsibility of the construct, rather than the pattern (if I understand it correctly, this `FindInMap` stuff gets added regardless of whether the user overrides the defaults or not).

Reply:
Yeah, good shout. I'll get rid of it as it was already copied from the `asg` tests.